src/Compilers/CSharp/Portable/Binder/Semantics/Operators/BinaryOperatorSignature.cs

```diff
@@ -70,5 +71,35 @@ public override int GetHashCode()
                    Hash.Combine(RightType,
                    Hash.Combine(Method, (int)Kind))));
         }
+
+        public RefKind LeftRefKind
+        {
+            get
+            {
+                Debug.Assert(
+                    Method != null &&
+                    Method.ParameterCount == 2 &&
+                    (Method.ParameterRefKinds.IsDefaultOrEmpty || Method.ParameterRefKinds.Length == 2));
+
+                return Method.ParameterRefKinds.IsDefaultOrEmpty
+                    ? RefKind.None
+                    : Method.ParameterRefKinds[0];
+            }
+        }
+
+        public RefKind RightRefKind
+        {
+            get
+            {
+                Debug.Assert(
+                    Method != null &&
```

Comment:
>Method != null && [](start = 20, length = 17)

Same comment. Consider passing ref-kinds to the constructor explicitly, the same way as  we do with types.

Reply:
As an alternative, we could always return None when ```Method``` is null.

---
In reply to: [155020652](https://github.com/dotnet/roslyn/pull/23122#discussion_r155020652) [](ancestors = 155020652)

Reply:
Will take into account null methods (built-in language operators that have no method symbol).

---
In reply to: [155021847](https://github.com/dotnet/roslyn/pull/23122#discussion_r155021847) [](ancestors = 155021847,155020652)